### PR TITLE
TESTS-111: Create unit tests for the Load, PV and SmartMeter states

### PR DESCRIPTION
**Pull request by ERafaelMartinez**
This PR contains unit tests for `LoadState()`, `PVState()` and `SmartMeterSate()` classes.

### CODE REVIEW

**Review by Hassan754 — DISMISSED** [review]
LGTM

File: tests/test_state/test_load_state.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @pytest.mark.parametrize(
+        "desired_energy, traded_energy, measured_energy_consumption, expected_deviation",
+        [(2000.0, 1000.0, 1.8, 0.8), (2000.0, 2000.0, 1.8, -0.2), (2000.0, 2000.0, 2.0, 0.0)])
+    def test_set_energy_measurement_kWh_unsettled_energy_calculation(
+            self, desired_energy, traded_energy, measured_energy_consumption, expected_deviation):
+        """Test different case scenarios for unsettled energy:
+            1st overproduction,
+            2nd underproduction,
+            3rd no unsettled energy.
```
Comment (hannesdiedrich) [inline tests/test_state/test_load_state.py:36]: Probably copy paste issue: `production` does not make sense for the load.

**Review by hannesdiedrich — COMMENTED** [review]
One minor comment, LGTM once addressed

File: tests/test_state/test_load_state.py (hunk 1 above)
Comment (ERafaelMartinez) [reply tests/test_state/test_load_state.py]: Thanks! Is now fixed.

**Comment by ERafaelMartinez** [comment]
@Hassan754 @hannesdiedrich 
Adapted to comments. Appreciate your re-review/approve.

**Review by Hassan754 — APPROVED** [review]

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**